components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -451,12 +453,34 @@ public static String getServerURL(String urlContext, boolean addProxyContextPath
                                       boolean addTenantQueryParamInLegacyMode, boolean addTenantPathParamInLegacyMode)
             throws IdentityRuntimeException {
 
-        URLResolverService urlResolverService = IdentityCoreServiceComponent.getURLResolverService();
+        return getAbsoluteURL(urlContext, addProxyContextPath, addWebContextRoot, addTenantQueryParamInLegacyMode,
+                addTenantPathParamInLegacyMode);
+    }
+
+    public static String getAbsoluteURL(String urlContext, boolean addProxyContextPath, boolean addWebContextRoot,
+                                        boolean addTenantQueryParamInLegacyMode, boolean addTenantPathParamInLegacyMode)
+            throws IdentityRuntimeException {
+
+        ServiceURLBuilder serviceURLBuilder = IdentityCoreServiceComponent.getServiceURLBuilder();
```

Comment:
Can't we call, ServiceURLBuilder.create()

Reply:
Intention of builder pattern is to write this code with less lines like,
ServiceURLBuilder.create().addPath().addParameter()

the respective API should validate for empty and null values

Reply:
The builder components doesn't take in any parameters. So, for example, ServiceURLBuilder.create().addPath().addTenantQueryParam() gets called, the ServiceURLBuilder registers this as addTenantQueryParam = true

Since we don't know the value of the booleans in getServerUrl method, we have to perform a check for the boolean parameters before adding it to the builder.